source/common/tls/connection_info_impl_base.cc

```diff
@@ -380,6 +380,17 @@ const std::string& ConnectionInfoImplBase::subjectPeerCertificate() const {
   });
 }
 
+const Ssl::ParsedX509NamePtr& ConnectionInfoImplBase::parsedSubjectPeerCertificate() const {
+  return getCachedValueOrCreate<Ssl::ParsedX509NamePtr>(
```

Comment:
Right here is where you can convert from unique_ptr to OptRef depending on if it's nullptr or not.